testng-core/src/test/java/test/support/SafeRandoms.java

```diff
@@ -0,0 +1,23 @@
+package test.support;
+
+import java.security.SecureRandom;
+import java.util.Random;
+
+public final class SafeRandoms {
+
+  private static final Random random = new SecureRandom();
```

Comment:
`SecureRandom` would be hard to reproduce in case of failure. It might be better using regular random and print the seed in case the test fails, so it could be reproduced locally.

Reply:
Ok. I have changed this to be `Random`.